Use KubernetesHook to create api client in KubernetesPodOperator (#20578)
mistercrunch/Airflow: airflow/providers/cncf/kubernetes/CHANGELOG.rst

Changelog
---------

main
....

Features
~~~~~~~~

KubernetesPodOperator now uses KubernetesHook
`````````````````````````````````````````````

Previously, KubernetesPodOperator relied on core Airflow configuration (namely setting for kubernetes executor) for certain settings used in client generation.  Now KubernetesPodOperator uses KubernetesHook, and the consideration of core k8s settings is officially deprecated.

> If you are using the Airflow configuration settings (e.g. as opposed to operator params) to configure the kubernetes client, then prior to the next major release you will need to add an Airflow connection and set your KPO processes to use that connection.

Reviewer: ```suggestion
If you are using the Airflow configuration settings (e.g. as opposed to operator params) to configure the kubernetes client, then prior to the next major release you will need to add an Airflow connection and set your KPO tasks to use that connection.
```

nit

* Enum-like class ``PodStatus`` is renamed ``PodPhase``, and the values are no longer lower-cased.

Features
~~~~~~~~

> * Can now derive kubernetes creds using airflow connection, through KubernetesHook

Reviewer 2: ```suggestion
* Can now derive Kubernetes credentials from an Airflow connection through the KubernetesHook
```
